cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/BuiltInRolesAdministrationTest.scala

```diff
@@ -0,0 +1,329 @@
+            |Secondly, the following query need to be executed:""".stripMargin)
+        query("GRANT ACCESS ON DEFAULT DATABASE TO PUBLIC", ResultAssertions((r) => {
+          assertStats(r, systemUpdates = 1)
+        })) {
+          statsOnlyResultTable()
+        }
+
+        p("The resulting `PUBLIC` role has the same privileges as the " +
+          "<<administration-built-in-roles-public-privileges,original built-in `PUBLIC` role>>.")
+
+//        query("SHOW ROLE PUBLIC PRIVILEGES", ResultAssertions(p => true)) {
+//          resultTable()
+//        }
+      }
+    }
+    section("The `reader` role", "administration-built-in-roles-reader", "enterprise-edition") {
+      p("The `reader` can perform read-only queries on all databases except `system`.")
+      section("Privileges of the `reader` role", "administration-built-in-roles-reader-privileges", "enterprise-edition") {
+        query("SHOW ROLE reader PRIVILEGES", ResultAssertions(p => true)) {
+          resultTable()
+        }
+      }
+      section("How to recreate the `reader` role", "administration-built-in-roles-reader-recreate", "enterprise-edition") {
+        initQueries(
+          "DROP ROLE reader")  // setup so that later when the grant query gets executed, it will show systemUpdates: 1
+        p(
+          """
+            |To restore the role to its original capabilities two steps are needed: First, if not already done, execute `DROP ROLE reader`.
+            |Secondly, the following queries need to be executed:""".stripMargin)
+
+        query("CREATE ROLE reader", ResultAssertions((r) => {
+          assertStats(r, systemUpdates = 1)
+        })) {
+          statsOnlyResultTable()
+        }
+        query("GRANT ACCESS ON DATABASE * TO reader", ResultAssertions((r) => {
+          assertStats(r, systemUpdates = 1)
+        })) {
+          statsOnlyResultTable()
+        }
+        query("GRANT MATCH {*} ON GRAPH * NODES * TO reader", ResultAssertions((r) => {
+          assertStats(r, systemUpdates = 1)
+        })) {
+          statsOnlyResultTable()
+        }
+        query("GRANT MATCH {*} ON GRAPH * RELATIONSHIPS * TO reader", ResultAssertions((r) => {
+          assertStats(r, systemUpdates = 1)
+        })) {
+          statsOnlyResultTable()
+        }
+
+        p("The resulting `reader` role has the same privileges as the " +
+          "<<administration-built-in-roles-reader-privileges,original built-in `reader` role>>.")
+
+//        query("SHOW ROLE reader PRIVILEGES", ResultAssertions(p => true)) {
+//          resultTable()
+//        }
+      }
+    }
+    section("The `editor` role", "administration-built-in-roles-editor", "enterprise-edition") {
+      p("The `editor` can perform read and write operations on all databases except `system`, but cannot make new labels or relationship types.")
```

Comment:
Same about graph vs. database as for reader.

Also you should mention property keys together with labels and relationship types. I don't know whether the concept of `token`  is well-defined and enough used in docs to make sense here or if it just is confusing to mention.